app.py

```diff
@@ -1,53 +1,146 @@
 import streamlit as st
 import pandas as pd
-import time
-
-# def loader():
-#     latest_iteration = st.empty()
-#     bar = st.progress(0)
+import json
+import data_upload_DHIS2 as dhis2
 
 # Initializing variables
 form_types = ("ICCM", "OPD", "RHGynobs", "Vaccination")
+# Ideally, these could be set in this app?
+# dhis2_username = ''
+# dhis2_password = ''
+# DHIS2_Test_Server_URL = ''
 
-st.title("MSF OCR Tool")
 
-st.write("### File Upload ###")
+# Function definitions!
+# """ Creates a list of dataValues by going through each cell
+# and recording the column, header, and cell value of non-None
 
-# results = ocr.run()
+# :param df: an iterable of data frames
+# :return: a list of JSON-ready dataValues
+# """
+def data_values(dfs):
+    data_values = []
+    for df in dfs:
+        columns = df.columns.tolist()
+        for col, i in df.iterrows():
+            for cell in columns:
+                if i[cell] is not None:
+                    new_cell = {"dataElement": f"{col}/{cell}", "value": f"{i[cell]}"}
+                    data_values.append(new_cell)
+    return data_values
+            
+# @st.cache_data
+# Ideally this would cache data but it messes with filling in info
+# """ Creates a JSON payload through the form inputs and, if not empty,
+# calling the getUID command to find the proper ID.
+
+# :param df: an iterable of data frames
+# :return: the JSON payload
+# """
+def convert_df(dfs):
+    json_export = {}
+    
+    if data_set == "":
+        data_set_id = ""
+    else:
+        data_set_id = dhis2.getUID("dataSets", data_set)
+    json_export["dataSet"] = f"{data_set_id}"
+    
+    json_export["period"] = f"{period_start}P7D"
+    
+    if org_unit == "":
+        org_unit_id = ""
+    else:
+        org_unit_id = dhis2.getUID("organisationUnits", org_unit)
+        
+    json_export["orgUnit"] = f"{org_unit_id}"
+    
+    json_export["dataValues"] = data_values(dfs.values())
+    return json.dumps(json_export)
 
+
+# Initial Display
+st.title("MSF OCR Tool")
+st.write("### File Upload ###")
 tally_sheet = st.file_uploader(
     "Please upload one or more images of a tally sheet",
     type=["png", "jpg", "jpeg"],
     accept_multiple_files=True

```

Comment:
I'm surprised we want to accept multiple files here, I thought most of the tally sheets were only one page, but now that I'm looking back at them, I'm realizing that isn't true. It could be a challenge for the system if users upload things in a different order than normal. No code changes needed for now, but example to add to our list for testing later.